Configuration/Skimming/src/DisappearingMuonsSkimming.cc

```diff
@@ -0,0 +1,413 @@
+          if (dR < staMinDr) {
+            staMinDr = dR;
+          }
+          //Pick the largest standalone muon within the cone
+          if (dR < minStandaloneDr_ && staDE > staMinDEoverE) {
+            staMinDEoverE = staDE;
+          }
+        }
+      }
+      if (staMinDr > minStandaloneDr_) {
+        totalRegion = true;
+      } else {
+        if (staMinDEoverE < maxStandaloneDE_) {
+          if (MuonTrackMass < minSigInvMass_ || MuonTrackMass > maxSigInvMass_) {
+            offPeak = true;
+          } else {
+            partialRegion = true;
+          }
+        }
+      }
+      nMuonTrackCand++;
+    }
+  }
+
+  if (nMuonTrackCand == 0)
+    return false;
+  bool passes = false;
+  //Pass all same sign CR events
+  if (sameSign && keepSameSign_) {
+    passes = true;
+  }
+  //Pass all total disappearance events
+  else if (totalRegion && keepTotalRegion_) {
+    passes = true;
+  }
+  //Pass all partial disappearkance off-peak events
+  else if (offPeak && keepOffPeak_) {
+    passes = true;
+  }
+  //Pass partial region events that pass minimum standalone muon DE/E.
+  else if (partialRegion && keepPartialRegion_) {
+    passes = true;
+  }
+  return passes;
+}
+
+bool DisappearingMuonsSkimming::passTriggers(const edm::Event& iEvent,
+                                             edm::EDGetToken m_trigResultsToken,
+                                             std::vector<std::string> m_muonPathsToPass) {
+  bool passTriggers = false;
+  edm::Handle<edm::TriggerResults> triggerResults;
+  iEvent.getByToken(m_trigResultsToken, triggerResults);
+  const edm::TriggerNames& trigNames = iEvent.triggerNames(*triggerResults);
+  for (size_t i = 0; i < trigNames.size(); ++i) {
+    const std::string& name = trigNames.triggerName(i);
+    for (auto& pathName : m_muonPathsToPass) {
+      if ((name.find(pathName) != std::string::npos)) {
+        if (triggerResults->accept(i)) {
+          passTriggers = true;
+        }
+      }
+    }
+  }
+  return passTriggers;
+}
+
+double DisappearingMuonsSkimming::getTrackIsolation(const edm::Event& iEvent,
+                                                    edm::Handle<reco::VertexCollection> vtxHandle,
+                                                    std::vector<reco::Track>::const_iterator& iTrack) {
+  bool foundtrack = false;
+  unsigned int vtxindex = -1;
+  unsigned int trackindex = -1;
+  double Isolation = 0;
+  for (unsigned int i = 0; i < vtxHandle->size(); i++) {
+    reco::VertexRef vtx(vtxHandle, i);
+    if (!vtx->isValid()) {
+      continue;
+    }
+    for (unsigned int j = 0; j < vtx->tracksSize(); j++) {
+      double dR = deltaR(vtx->trackRefAt(j)->eta(), vtx->trackRefAt(j)->phi(), iTrack->eta(), iTrack->phi());
+      double dPt = fabs(vtx->trackRefAt(j)->pt() - iTrack->pt()) / iTrack->pt();
+      //Find the vertex track that is the same as the probe
+      if (dR < 0.001 && dPt < 0.001) {
```

Comment:
I'm confused, why can't we just compare the refs here?

Reply:
I'd like to compare the refs directly, but I can't quite figure out how to convert the TrackBaseRef to a reco::Track (or vice versa) to allow the comparison. For now I implemented the dPt followed by dR2 check.